File: 3105/sol.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  int longestMonotonicSubarray(vector<int>& nums) {
    int res = 1, len = 1;
    for (int i = 1; i < nums.size(); i++){
      len = nums[i] > nums[i-1] ? len+1 : 1;
      res = max(res, len);
    }
    len = 1;
    for (int i = 1; i < nums.size(); i++){
      len = nums[i] < nums[i-1] ? len+1 : 1;
      res = max(res, len);
    }
    return res;
  }
};

File: 2401/sol.py
class Solution(object):
  def longestNiceSubarray(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    start, maxLength, mask = 0, 0, 0
    for i in range(len(nums)):
      while mask & nums[i] != 0:
        mask ^= nums[start]
        start += 1
      mask |= nums[i]
      maxLength = max(maxLength, i-start+1)
    return maxLength
        